### PR TITLE
fix issue with loop out of array size

### PR DESCRIPTION
**Pull request by cecilecaillol**
#### PR description:

This addresses issue https://github.com/cms-sw/cmssw/issues/37659 (loop going out of array size)

### CODE REVIEW

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

File: L1Trigger/TrackTrigger/src/L1TrackQuality.cc [hunk 1]
```diff
@@ -118,12 +118,14 @@ std::vector<float> L1TrackQuality::featureTransform(TTTrack<Ref_Phase2TrackerDig
...
   int n_bendchi2 = static_cast<int>(bendchi2_bins.size());
   float tmp_trk_bendchi2_bin = -1;
-  for (int i = 0; i < n_bendchi2; i++) {
+  for (int i = 0; i < (n_bendchi2 - 1); i++) {
     if (tmp_trk_bendchi2 >= bendchi2_bins[i] && tmp_trk_bendchi2 < bendchi2_bins[i + 1]) {
       tmp_trk_bendchi2_bin = i;
       break;
     }
   }
+  if (tmp_trk_bendchi2 >= bendchi2_bins[n_bendchi2 - 1])
```
Comment (perrotta) [inline L1Trigger/TrackTrigger/src/L1TrackQuality.cc:127]: This moves all the overflows (tmp_trk_bendchi2 > 50.) into the 7th bin: was it intended?

File: L1Trigger/TrackTrigger/src/L1TrackQuality.cc [hunk 2]
```diff
@@ -118,12 +118,14 @@ std::vector<float> L1TrackQuality::featureTransform(TTTrack<Ref_Phase2TrackerDig
...
+    tmp_trk_bendchi2_bin = n_bendchi2 - 1;
   if (tmp_trk_bendchi2_bin < 0)
```
Comment (perrotta) [inline L1Trigger/TrackTrigger/src/L1TrackQuality.cc:129]: This moves all the underflows (`tmp_trk_bendchi2 < 0.`) into the 8th bin: was it intended?

**Review by perrotta — COMMENTED** [review]
The very same structure repeats identically in the similar parts below (for `tmp_trk_chi2rphi`, `tmp_trk_chi2rz`) if this needs to be fixed, the same also holds for the other ones.

File: L1Trigger/TrackTrigger/src/L1TrackQuality.cc (hunk 2 above)
Comment (cecilecaillol) [reply L1Trigger/TrackTrigger/src/L1TrackQuality.cc:129]: @skinnari Same here

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
@skinnari kind reminder

**Comment by skinnari** [comment]
@cgsavard @Jingyan95  can you please help check these changes? Thanks!

File: L1Trigger/TrackTrigger/src/L1TrackQuality.cc (hunk 1 above)
Comment (cgsavard) [reply L1Trigger/TrackTrigger/src/L1TrackQuality.cc:127]: @perrotta @cecilecaillol The fix on L121 is good (loop out of array size) but L127 and L128 should be removed. We want all of the overflow to go into the 8th bin which is 50 - inf. L129 and L130 do this already by assigning it bin 8 if the bin has not been selected yet.

File: L1Trigger/TrackTrigger/src/L1TrackQuality.cc (hunk 2 above)
Comment (cgsavard) [reply L1Trigger/TrackTrigger/src/L1TrackQuality.cc:129]: Same as above, L137 and L153 are good but L143-144 and L159-160 should be removed.

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

**Comment by cecilecaillol** [comment]
@AdrianoDee, @srimanob, Could you please review this short PR?

**Comment by AdrianoDee** [comment]
+upgrade
- Technical PR
- solving https://github.com/cms-sw/cmssw/issues/37659

**Comment by perrotta** [comment]
+1

**Pull request merged**